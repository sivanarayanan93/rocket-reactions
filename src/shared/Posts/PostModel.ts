import { getReactions } from "../Reactions/ReactionsApi";

export const getPosts = () => {
  return [{id: 1, content: 'Content 1'},{id: 2, content: 'Content 2'}];
}

export const getReactionsForPost = async() => {
  try {
    const res = await getReactions();  
    return res && res.data ? res.data : [];
  } catch (err) {
    return [];
  }
}