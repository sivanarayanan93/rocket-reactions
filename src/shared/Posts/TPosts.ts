import { TStringOrNumber } from "../common/TCommon";

export type TPost = {
 id: TStringOrNumber,
 content: string
}

export type TPosts = TPost[];
