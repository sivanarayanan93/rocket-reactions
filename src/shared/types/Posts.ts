import { TStringOrNumber } from "../common/TCommon";

export type Post = {
 id: TStringOrNumber,
 content: string
}

export type Posts = Post[];
