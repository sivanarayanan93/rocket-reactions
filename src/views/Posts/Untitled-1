if (reactionId && !lastAddReactionPromise.current[reactionId]) {

      // if (!lastAddReactionPromise.current) {
        const {id, contentId, userId, emoji, contentReactionId} = selectedReaction;
      
        const currentPromise = contentReactionId ? deleteReaction(contentReactionId) : addReaction({id, contentId, userId});

        lastAddReactionPromise.current[reactionId] = { currentPromise, reaction: selectedReaction };

        currentPromise.then((res) => {
          let response = res && res.data
          // if (lastAddReactionPromise.current[response.reaction_id].currentPromise === currentPromise && response) {

            const newContentReactionId = contentReactionId ? contentReactionId : res.data && response.id;

            let currentReaction = { ...lastAddReactionPromise.current[response.reaction_id].reaction}

            const oldId = currentReaction.id,
            oldEmoji = currentReaction.emoji,
            oldContentReactionId = currentReaction.contentReactionId;

            successReactionCallback({id: oldId, emoji: oldEmoji, contentReactionId: newContentReactionId}, !oldContentReactionId);
            delete lastAddReactionPromise.current[reactionId]
          // }
        }).catch(() => {
          if (lastAddReactionPromise.current[reactionId].currentPromise === currentPromise) {
            setSelectedReaction({});
            delete lastAddReactionPromise.current[reactionId]
          }
        });
      }