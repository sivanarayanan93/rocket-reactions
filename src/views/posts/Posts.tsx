import React from 'react'
import Post from './Post';
import { Post as PostType } from '../../shared/types/Posts';

type Posts = {
  contents: PostType[]
}

const Posts: React.FC<Posts> = ({ contents }) => {
  return (
    <>
      {contents && contents.map((content) => (
        <Post key={content.id} post={content}/>
      ))}
    </>
  )
}

export default Posts
