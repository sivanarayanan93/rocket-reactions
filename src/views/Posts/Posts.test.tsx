import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../../store';
import Posts from './index';

describe('Posts component', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Provider store={store}> <Posts contents={[{id: 1, content: 'Content 1'},{id: 2, content: 'Content 2'}]}/></Provider>);
    expect(asFragment()).toMatchSnapshot();
  });
});