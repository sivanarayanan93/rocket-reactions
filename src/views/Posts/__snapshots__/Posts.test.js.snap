// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Posts component matches snapshot 1`] = `
<DocumentFragment>
   
  <h2>
    Posts
  </h2>
  <div
    class="sc-kEqXSa kdUyVv"
  >
    <div
      class="sc-iqAclL jNboji"
    >
      Content 1 - 1
    </div>
    <div
      class="sc-bdnxRM juMqzC"
    >
      <div
        class="added-reactions"
      />
      <button
        aria-expanded="false"
        aria-haspopup="true"
        class="sc-jrsJWt kDvBKi"
        data-toggle="dropdown"
        style="font-weight: 600;"
      >
        R
      </button>
    </div>
  </div>
  <div
    class="sc-kEqXSa kdUyVv"
  >
    <div
      class="sc-iqAclL jNboji"
    >
      Content 2 - 2
    </div>
    <div
      class="sc-bdnxRM juMqzC"
    >
      <div
        class="added-reactions"
      />
      <button
        aria-expanded="false"
        aria-haspopup="true"
        class="sc-jrsJWt kDvBKi"
        data-toggle="dropdown"
        style="font-weight: 600;"
      >
        R
      </button>
    </div>
  </div>
</DocumentFragment>
`;
